### ISSUE TITLE
[question] proper type annotation for __dict__

### ISSUE THREAD

**Issue by AMDmi3** [python/mypy#6523]
I'm wondering which is the correct way of annotating `__dict__` property. `Dict[str, Any]` seem quite obvious and is used multiple times in typeshed, but:

```python
from typing import Any, Dict

class Test:
    @property
    def __dict__(self) -> Dict[str, Any]:
        return {}
```
```
% mypy test.py 
test.py:5: error: Signature of "__dict__" incompatible with supertype "object"
```

**Comment by ethanhs** [comment]
This appears to be a duplicate of https://github.com/python/mypy/issues/5836. As a workaround in your case, you can likely do something like:

```python
from typing import Dict, Any, TYPE_CHECKING

class Test:
    if TYPE_CHECKING:
        __dict__ = {}  # type: Dict[str, Any]
    else:
        @property
        def __dict__(self) -> Dict[str, Any]:
            return {}
```

and mypy should be happy.

**Comment by nyuszika7h** [comment]
This does not seem to be fixed in mypy 0.812 (Python 3.9.4). Am I missing something here?
